.gitlab/e2e_install_packages/common.yml

```diff
@@ -50,6 +50,16 @@
     - export DATADOG_AGENT_API_KEY=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $INSTALL_SCRIPT_API_KEY )
     - inv -e new-e2e-tests.run --targets $TARGETS --junit-tar "junit-${CI_JOB_ID}.tgz" ${EXTRA_PARAMS} --src-agent-version $START_MAJOR_VERSION --dest-agent-version $END_MAJOR_VERSION
 
+.new-e2e_script_persisting_integrations:
+  stage: e2e_install_packages
+  variables:
+    TARGETS: ./tests/agent-platform/persisting-integrations
+    TEAM: agent-delivery
+    EXTRA_PARAMS: --osversion $E2E_OSVERS --platform $E2E_PLATFORM --arch $E2E_ARCH --flavor $FLAVOR
+  script:
+    - export DATADOG_AGENT_API_KEY=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $INSTALL_SCRIPT_API_KEY )
```

Comment:
```suggestion
    - DATADOG_AGENT_API_KEY=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $INSTALL_SCRIPT_API_KEY ) || exit $?; export DATADOG_AGENT_API_KEY
```
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏For now, we need to fetch secret like this to get the proper error to raise in the CI

Reply:
Nice catch @amenasria ! I remember seeing a PR about that, I guess I copied the old changes because the branch is behind.
However, we don't see the new-e2e test on the CI? When are the upgrade tests executed?

Reply:
Addressed in #29760